Added a monster statblock note template

felix-schneider opened the pull request:
It's based on [statblock5e](https://valloric.github.io/statblock5e/)'s appearance.

Unfortunately, the border gets very badly messed up, just like for the other notes. I tried fixing it with constant offsets, but it seems that the shifted components scale with the height of the note!

For example, the shadow on the side of the regular note and my monster note moves up further the bigger the note contents are. I kind of gave up on fixing it after that... This was tested with miktex on Windows 10.

felix-schneider committed:
Added a monster statblock note template

Artefact2 commented:
Can you add an example in the demo document ? Just to show how one is supposed to use it.

> Unfortunately, the border gets very badly messed up, just like for the other notes.

Which of the other notes have bad borders?

felix-schneider committed:
Added Example

felix-schneider commented:
I've amended the example.

Here are a few examples of the misaligned boundaries:
![artifact1](https://user-images.githubusercontent.com/208336/32452754-bbe7ed3e-c31a-11e7-8a20-30735a8cd22a.png)
This is for the variant note, where it is the least noticeable.

![artifact2](https://user-images.githubusercontent.com/208336/32452775-ca81867a-c31a-11e7-8430-b7f575c957ef.png)
This is the corner of a wide variant note. See how the top border sticks out more for the wider note. The offset actually scales with the size of the note.

![artifact3](https://user-images.githubusercontent.com/208336/32452822-e4fd9ea8-c31a-11e7-8fe9-0d330bb99705.png)
This is the default note from the example as compiled by Miktex on windows.

![artifact4](https://user-images.githubusercontent.com/208336/32452859-03e86758-c31b-11e7-8368-50593e1615a9.png)
Finally, this is a test by me showing how the offset is different for different sizes of notes.

Artefact2 commented:
Interesting, I can't reproduce any of these issues on my end. Could it come from your pdf reader, or from your tex distribution?

![](https://i.imgur.com/16IjMUS.png)

![](https://i.imgur.com/Iu28JBM.png)

I use texlive 2017 and zathura with mupdf backend. Can you see the same artefacts on this sample document? https://drive.google.com/open?id=1XVm1a7onacZjPLqRHW4m-eDIoBTWCBh8

felix-schneider commented:
No, your sample document does not have this issue. Seems to be on my end then.

I would suggest moving the corners of the default note slightly outwards, so that the shadow for the main note body does not extend beyond the end of the corners. But that's just an aside, it does not relate to this PR.

felix-schneider committed:
Adjusted border thickness and overlap

felix-schneider commented:
I've changed to texlive, the notes now render correctly so long as they are multiline, which is fine.
![artifact5](https://user-images.githubusercontent.com/208336/32469840-c9d147e4-c354-11e7-8c59-aecf149939d6.png)

Using that, I have adjusted spacing for the monster statblock note and it should now be ready for use.

Artefact2 commented:
I've merged your commits, thanks for the contribution!

If you want to add your name/email for copyright info in the relevant files, please do so. (I assume you're releasing the code/assets under WTFPLv2 ?)
